### PR TITLE
Run  'pip check' in tests

### LINKED ISSUE #4: `pip check` fails with pip==24.2: upf-to-json is not supported on this platform
### Issue

Running a `pip check` in a fresh conda environment with latest pip==24.2 reports a problem that's likely caused by invalid wheel metadata. More info and discussion can be found on this issue:
https://github.com/pypa/pip/issues/11054

To reproduce
```
$ conda create -n test pip=22.4 upf-to-json
$ conda activate test
$ pip check
upf-to-json 0.9.2 is not supported on this platform
```

Note that `pip check` does not fail if the `upf-to-json` package is installed with pip from PYPI.

### PR DESCRIPTION
**Pull request by danielhollas**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #4

### CODE REVIEW

**Comment by danielhollas** [comment]
@conda-forge-admin, please rerender

**Review by simonpintarelli — APPROVED** [review]
lgtm

**Comment by danielhollas** [comment]
I've dugged a bit more into the root cause of the problem described in #4, and noticed that version 0.9.2 still supported python 2.7. I've changed the minimum python version back to 2.7 in this PR, but perhaps a better solution instead of publishing a new build is to simply publish a new version of the package. I've opened #6 which integrates changes in this PR for version 0.9.5. @simonpintarelli if you want to release one of the earlier versions as well (0.9.4?) let me know and I can prepare PR for that instead. 

tl;dr. I've switched this PR to draft and I think it's simpler to just merge #6.
